src/amo/components/GuidesAddonCard/index.js

```diff
@@ -0,0 +1,222 @@
+/* @flow */
+import * as React from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+import AddonTitle from 'amo/components/AddonTitle';
+import AddonCompatibilityError from 'amo/components/AddonCompatibilityError';
+import { makeQueryStringWithUTM } from 'amo/utils';
+import AMInstallButton from 'core/components/AMInstallButton';
+import {
+  INCOMPATIBLE_NOT_FIREFOX,
+  INSTALL_SOURCE_GUIDES_PAGE,
+  UNKNOWN,
+} from 'core/constants';
+import { getAddonIconUrl } from 'core/imageUtils';
+import { withInstallHelpers } from 'core/installAddon';
+import translate from 'core/i18n/translate';
+import { getAddonByGUID } from 'core/reducers/addons';
+import { getVersionById } from 'core/reducers/versions';
+import { getErrorMessage } from 'core/utils/addons';
+import { getClientCompatibility } from 'core/utils/compatibility';
+import Card from 'ui/components/Card';
+import Icon from 'ui/components/Icon';
+import Button from 'ui/components/Button';
+import Notice from 'ui/components/Notice';
+import type { AddonType } from 'core/types/addons';
+import type { AddonVersionType } from 'core/reducers/versions';
+import type { AppState } from 'amo/store';
+import type { UserAgentInfoType } from 'core/reducers/api';
+import type { I18nType } from 'core/types/i18n';
+import type { LocationType } from 'core/types/router';
+import type { WithInstallHelpersInjectedProps } from 'core/installAddon';
+
+import './styles.scss';
+
+type Props = {
+  addonGuid: string,
```

Comment:
Instead of mapping `addonGuid` to an `addon` in state, I'd like to see you try accepting an `addon` prop directly. The reason is because `GuidesAddonCard` is not responsible for loading an add-on, `pages/Guides` is responsible for that. It's dangerous to mix responsibilities like this because it can lead to subtle bugs that are hard to track down. Also, it's going to make `GuidesAddonCard` and its tests a lot simpler!

The biggest change you will have to make is in `pages/Guides`. Instead of [passing `addonGuid`](https://github.com/mozilla/addons-frontend/pull/6859/files#diff-6bda2327e05b9cda8e014f48186e91e8R138) you'll need to map each guid to its add-on within state.

Before writing this comment I started to request some test cleanup related to loading an add-on into state and passing a guid. You can ignore those comments but I'll leave them so that you understand that pattern for a future time when you need it.

Reply:
I did do this initially but let me take another look.

Reply:
Oh? Did you run into problems with it?

Reply:
with the latest change I tried revisiting @willdurand's initially suggestion. let me know if this is okay